src/openvic-simulation/map/Map.cpp

```diff
@@ -644,3 +646,89 @@ bool Map::generate_and_load_province_adjacencies(std::vector<LineObject> const&
 	);
 	return ret;
 }
+
+bool Map::load_climate_file(ModifierManager const& modifier_manager, ast::NodeCPtr root) {
+	bool ret = expect_dictionary_reserve_length(climates, [this, &modifier_manager](std::string_view identifier, ast::NodeCPtr node) -> bool {
+		if (identifier.empty()) {
+			Logger::error("Invalid climate identifier - empty!");
+			return false;
+		}
+
+		bool ret = true;
+		Climate* cur_climate = climates.get_item_by_identifier(identifier);
+		if (cur_climate == nullptr) {
+			ModifierValue values;
+			ret &= modifier_manager.expect_modifier_value(move_variable_callback(values))(node);
+			ret &= climates.add_item({ identifier, std::move(values) });
+		} else {
+			ret &= expect_list(expect_province_identifier([this, &cur_climate, &identifier](Province& province) {
```

Comment:
```suggestion
			ret &= expect_list_reserve_length(*cur_climate, expect_province_identifier([cur_climate, &identifier](Province& province) {
```
`Climate` works with `expect_list_reserve_length`, since it inherits a `reserve` function from `ProvinceSet`. Also `this` isn't needed in the lambda capture, and `cur_climate` is better off being captured by value rather than reference, since it's already a pointer so there's no need to reference to it (which is basically using a pointer to pointer) when you can just copy it and point directly to the climate.